src/libraries/Microsoft.Extensions.Logging.Console/src/AnsiParser.cs

```diff
@@ -0,0 +1,201 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.IO;
+
+namespace Microsoft.Extensions.Logging.Console
+{
+    internal class AnsiParser
+    {
+        private readonly Action<string, int, int, ConsoleColor?, ConsoleColor?> _onParseWrite;
+        public AnsiParser(Action<string, int, int, ConsoleColor?, ConsoleColor?> onParseWrite)
+        {
+            if (onParseWrite == null)
+            {
+                throw new ArgumentNullException(nameof(onParseWrite));
+            }
+            _onParseWrite = onParseWrite;
+        }
+
+        /// <summary>
+        /// Parses a subset of display attributes
+        /// Set Display Attributes
+        /// Set Attribute Mode [{attr1};...;{attrn}m
+        /// Sets multiple display attribute settings. The following lists standard attributes that are getting parsed:
+        /// 1 Bright
+        /// Foreground Colours
+        /// 30 Black
+        /// 31 Red
+        /// 32 Green
+        /// 33 Yellow
+        /// 34 Blue
+        /// 35 Magenta
+        /// 36 Cyan
+        /// 37 White
+        /// Background Colours
+        /// 40 Black
+        /// 41 Red
+        /// 42 Green
+        /// 43 Yellow
+        /// 44 Blue
+        /// 45 Magenta
+        /// 46 Cyan
+        /// 47 White
+        /// </summary>
+        public void Parse(string message)
```

Comment:
This needs a bunch of tests

Reply:
I added more tests and also working on making the parser more robust for messages with invalid codes. perhaps that could go on a next PR

Reply:
CI failure was unrelated. I just pushed up a commit that fixes up issues with the parser making it more robust for messages with invalid code.